components/board/fake/board.go

```diff
@@ -238,26 +237,37 @@ func (b *Board) StreamTicks(ctx context.Context, interrupts []board.DigitalInter
 	extra map[string]interface{},
 ) error {
 	for _, i := range interrupts {
-		AddCallback(i.(*DigitalInterruptWrapper), ch)
+		name := i.Name()
+		d, ok := b.Digitals[name]
+		if ok {
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			case ch <- board.Tick{Name: d.conf.Name, High: randomBool(), TimestampNanosec: uint64(time.Now().Unix())}:
+			default:
+				// if nothing is listening to the channel just do nothing.
+			}
+		}
 	}
 	return nil
 }
 
+func randomBool() bool {
+	//nolint:gosec
```

Comment:
I did this also on a previous PR, but Alan suggested not ignoring this for potential security purposes? I think it's fine because we don't care what this random number is but yeah just a thought

Reply:
Good catch, Martha! At first glance I don't see why the linter is unhappy here, but `gosec` is usually security-based stuff, and it usually has good suggestions. Why does it think this line is dangerous?

Reply:
I looked into it briefly last time around, and I _think_ it's only dangerous if you need to use the resulting random number for a security purpose, maybe like a randomly generated key or something I'm not really sure. it seemed like this case did not apply but I wasn't fully understanding so I just changed it to the more secure way to be safe

Reply:
We're only using this function in a single place, right? I wonder if the linter would be happier if we inlined it.

Reply:
the linter says `Use of weak random number generator (math/rand instead of crypto/rand)` I could look into using `crypto/rand ` but we use `math/rand` with the nolint flag other places in our codebase.

Reply:
I can find you the example of crypto/rand because I previously implemented it in this file if you want but if we nolint elsewhere I think this is fine

Reply:
Ohh, thanks for explaining! The linter is concerned that it is easy for a malicious actor to predict what the future values coming from the RNG will be. I'd inline this function on line 246 (so it's clear what the random values are used for), then put in a comment explaining that's what the linter is worried about, but we don't care because it doesn't matter if someone could predict whether the fake interrupts will be initialized to high or low, and then leave the nolint directive.